tests/test_fieldtypes.py

```diff
@@ -526,10 +527,18 @@ def test_digest():
         excinfo.match(r".*Invalid MD5.*")
 
 
+if PY_312:
+    # Starting from Python 3.12, pathlib._Flavours are removed as you can now properly subclass pathlib.Path
+    # See also: https://github.com/python/cpython/issues/88302
+    PosixFlavour = pathlib.PurePosixPath
```

Comment:
The `_flavour` property in 3.12 is a link to the `posixpath` or `ntpath` module. We could do something similar by having a function `def custom_flavour(sep, altsep)` that yields a `patch()`ed posixpath with the given separator and alt separator and modify `PureCustomPath` like:
```
    class PureCustomPath(pathlib.PurePath):
        if PY_312:
            _flavour = custom_flavour(sep, altsep)
        else:
            _flavour = CustomFlavour()
```

Our tests however just need a PurePath subclass instance with a `_flavour` property that has `sep` and `altsep` properties, so a simpler setup would for this case also work.

Reply:
i'm happy to accept any suggestions

Reply:
I'd say, just do
```suggestion
    PosixFlavour = Mock
```

as we probably don't want to really mock `posixpath` and now it is still clear that flavour is something different.

Reply:
This change gives me an error:

```python
.tox/py3/lib/python3.12/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
tests/test_fieldtypes.py:558: in <module>
    (custom_pure_path(sep="/", altsep="")("/foo/bar"), True),
tests/test_fieldtypes.py:547: in custom_pure_path
    class PureCustomPath(pathlib.PurePath):
tests/test_fieldtypes.py:548: in PureCustomPath
    _flavour = CustomFlavour()
tests/test_fieldtypes.py:543: in __new__
    instance.sep = sep
/usr/local/lib/python3.12/unittest/mock.py:771: in __setattr__
    elif (self._spec_set and self._mock_methods is not None and
/usr/local/lib/python3.12/unittest/mock.py:656: in __getattr__
    elif self._mock_methods is not None:
/usr/local/lib/python3.12/unittest/mock.py:655: in __getattr__
    raise AttributeError(name)
E   AttributeError: _mock_methods
```

Reply:
@pyrco any ideas?